src/common/ceph_context.cc

```diff
@@ -254,9 +254,11 @@ class CephContextObs : public md_config_obs_t {
 	conf->enable_experimental_unrecoverable_data_corrupting_features,
 	cct->_experimental_features);
       ceph_spin_unlock(&cct->_feature_lock);
+#if defined(NDEBUG)
```

Comment:
NDEBUG is defined for non-DEBUG builds, so if we want to respect this macro, better of putting
```c
#ifndef NDEBUG
```

and i believe this warning is intentional, and helpful to our end users. it makes sure they are aware of the risk of using experimental features.